source/MaterialXCore/Document.cpp

```diff
@@ -608,11 +608,19 @@ void Document::upgradeVersion()
         if (interfaceElem && interfaceElem->hasType())
         {
             string type = interfaceElem->getAttribute(TypedElement::TYPE_ATTRIBUTE);
-            if (type != MULTI_OUTPUT_TYPE_STRING)
+            OutputPtr outputPtr;
+            if (type != MULTI_OUTPUT_TYPE_STRING && type != NONE_TYPE_STRING)
```

Comment:
I think adding a "none" type would be ok. Or am I missing something why the "none" type is special?

Reply:
"none" should mean there is no output. Not that the output type is "none". No idea what the later means? e.g. can you connect "none" to "none" - a weird hypothetical but code would need to be written to deal with it.